lib/brainstem/query_strategies/base_strategy.rb

```diff
@@ -42,17 +42,26 @@ def calculate_per_page
 
       private
 
-      def get_ids_sql(scope)
+      def get_models(scope)
+        ids = get_ids_for_page(scope)
+        id_lookup = {}
+        ids.each.with_index { |id, index| id_lookup[id] = index }
+        scope.klass.where(id: id_lookup.keys).sort_by { |model| id_lookup[model.id] }
+      end
+
+      def get_ids_for_page(scope)
         if use_calc_row?
+          # The paginator uses mysql SQL_CALC_FOUND_ROWS.
+          ids = @options[:paginator]&.get_ids_for_page(calculate_page, scope)
+          return ids if ids.present?
```

Comment:
Could we get away with not nesting `@options[:paginator]` inside `use_calc_row?`? We may be creating a situation where people provide a paginator, and it doesn't get used, because `use_calc_row?` is false for some reason.

Reply:
I went back and forth on this, and I agree with you. I went with the more cagey approach, but I'm happy to pull it out of the check, since the behavior of the paginator should be completely hidden from brainstem. It should be up to the creator of the paginator to make sure their runtime can handle their strategy.